aiofreepybox/aiofreepybox.py

```diff
@@ -1,114 +1,285 @@
+                    __name__.rsplit(".", 1)[0] + ".api", str(n) + ".py"
+                ),
+                api_l,
+            )
+        )
+        if len(api_l) != l:
+            raise KeyError
+
+        mods = {
+            str(n): import_module(".api." + str(n), package=__name__.rsplit(".", 1)[0])
+            for n in api_l
+        }
+        return mods
+
+    def __getattr__(self, k):
+        """ Return API attribute """
+
+        if "__" in k:
+            raise AttributeError
+
+        try:
+            mod = self.__imp_api__([k])
+        except KeyError:
+            _LOGGER.error(f"Invalid API name: {k}")
+            raise AttributeError
+        else:
+            if self._access:
+                kwargs: Dict[str, str] = {}
+                if k in _API_MODS_PARAMS and isinstance(_API_MODS_PARAMS[k], dict):
+                    kwargs = _API_MODS_PARAMS[k]
+                setattr(
+                    self, k, getattr(mod[k], k.capitalize())(self._access, **kwargs)
+                )
+            else:
+                return getattr(mod[k], k.capitalize())
+
+        return getattr(self, k)
+
+    def clean_db(self, uids: Optional[list] = None, all_: bool = False) -> bool:
+        """
+        Remove unauth fbx's from db
+
+        uids : `list`, optional
+            , Default to None
+        all_ : `bool`
+            , Default to False
+        """
+
+        if uids is None:
+            uids = [
+                base64.b64decode(x.name.rsplit("_", 1)[1]).decode("utf-8")
+                for x in Path(self.data_dir).glob(_F_DB_NAME + "_*")
+            ]
+        auth = [
+            base64.b64decode(x.name.rsplit("_", 1)[1]).decode("utf-8")
+            for x in Path(self.data_dir).glob(_F_TOKEN_NAME + "_*")
+        ]
+        c = 0
+        for uid in uids:
+            if uid not in auth or all_:
+                fname = Path(self.data_dir).joinpath(
+                    _F_DB_NAME
+                    + "_"
+                    + base64.b64encode(uid.encode("utf-8")).decode("utf-8")
+                )
+                try:
+                    fname.unlink()
+                    c += 1
+                except FileNotFoundError:
+                    pass
+        if c != 0:
+            _LOGGER.debug(f"{c} db entry cleared")
+            return True
+        return False
+
+    async def close(self) -> None:
+        """
         Close the freebox session
-        '''
-        if self._access is None:
-            raise NotOpenError('Freebox is not open')
+        """
+
+        if self._session and self._session.closed:  # type: ignore # noqa
+            return None
+
+        try:
+            await self._access.post("login/logout")  # type: ignore # noqa
+        except AttributeError:
+            pass
+        try:
+            await self._session.close()  # type: ignore # noqa
+            await asyncio.sleep(0.250)
+        except AttributeError:
+            pass
```

Comment:
Exactly, whatever the state (like losing the link before the end) it will always close without complaining, that's why i reduced the scope as much as i could (to AttributeError ) and added one try block per request.